### DISCUSSION TITLE
ProteomicsLFQ. Controlling output to MSstats.csv

### DISCUSSION THREAD

**Discussion by ivandatasci** [OpenMS/OpenMS#6661] [Code]
Running ProteomicsLFQ can produce a MSstats.csv file.

Is it possible to control the report on the ProteinName column?
Say I am more interested in Proteins rather than peptides or protein groups.
What should be changed?

I can show examples of both cases. Focus on first column only, please.

Instead of this:

| ProteinName                              | PeptideSequence    | PrecursorCharge | Condition | BioReplicate | Run |   Intensity | Reference        |
| ---------------------------------------- | ------------------ | --------------- | --------- | ------------ | --- | ----------- | ---------------- |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               2 | C1        |            3 |   3 |   6,478,549 | fs191119_03.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               3 | C1        |            1 |   1 |  82,376,180 | fs191119_01.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               3 | C1        |            2 |   2 |  73,607,360 | fs191119_02.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               3 | C1        |            3 |   3 | 118,140,000 | fs191119_03.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               3 | C2        |            4 |   4 |  18,171,230 | fs191119_04.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               3 | C2        |            5 |   5 | 102,160,400 | fs191119_05.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               3 | C2        |            6 |   6 |  88,341,210 | fs191119_06.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               4 | C1        |            1 |   1 | 183,583,800 | fs191119_01.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               4 | C1        |            2 |   2 | 168,417,100 | fs191119_02.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               4 | C1        |            3 |   3 | 232,127,000 | fs191119_03.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               4 | C2        |            4 |   4 |  21,564,360 | fs191119_04.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               4 | C2        |            5 |   5 | 247,948,700 | fs191119_05.mzML |
| ENSP00000342070.3;ENSP00000345672.5;E... | SGVYQHVTGEMMGGHAIR |               4 | C2        |            6 |   6 | 182,627,300 | fs191119_06.mzML |

I prefer this:

| ProteinName       | PeptideSequence    | PrecursorCharge | Condition | BioReplicate | Run         |     Intensity |
| ----------------- | ------------------ | --------------- | --------- | ------------ | ----------- | ------------- |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               2 | C1        |            1 | fs191119_01 |   3,690,920.8 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               2 | C1        |            2 | fs191119_02 |               |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               2 | C1        |            3 | fs191119_03 |   3,841,485.5 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               2 | C2        |            4 | fs191119_04 |   1,726,836.1 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               2 | C2        |            5 | fs191119_05 |               |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               2 | C2        |            6 | fs191119_06 |   2,706,018.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               3 | C1        |            1 | fs191119_01 | 103,319,040.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               3 | C1        |            2 | fs191119_02 |  88,561,064.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               3 | C1        |            3 | fs191119_03 | 146,085,808.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               3 | C2        |            4 | fs191119_04 |  19,869,756.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               3 | C2        |            5 | fs191119_05 | 132,566,400.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               3 | C2        |            6 | fs191119_06 | 113,543,776.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               4 | C1        |            1 | fs191119_01 | 231,161,440.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               4 | C1        |            2 | fs191119_02 | 218,904,400.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               4 | C1        |            3 | fs191119_03 | 290,255,712.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               4 | C2        |            4 | fs191119_04 |  27,907,836.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               4 | C2        |            5 | fs191119_05 | 317,969,088.0 |
| ENSP00000342070.3 | SGVYQHVTGEMMGGHAIR |               4 | C2        |            6 | fs191119_06 | 235,640,608.0 |

How should I tweak my procedure all the way from the searches to get the second type of report?

**Comment by jpfeuffer** [comment]
Hi!

There is no such functionality for IMO good reasons.

You did not mention if you want to just reduce to an arbitrary protein or if you want to duplicate rows. Your example looks more like reducing.

Reducing: Which one to pick? This will be completely arbitrary and you only have 1/n chances to pick the correct one. If it is just for display, fine. You will just need to do something like substr(table[,0], 0, firstIndexOf(';').

Duplicating: MSstats will consider them shared peptides and last time I checked MSstats removes them for quantification. Therefore it should be roughly the same as removing them.

HTH

**Reply by ivandatasci** [reply]
Got it.
I was thinking that, perhaps, there was functionality to pick the one protein out of the group, a pick that could be defended statistically or by taking some risk, something smarter that the 1/n chance that you mentioned above.

For example, to use the example in the table above, we could imagine that protein ENSP00000342070.3 has additional matching peptides but ENSP00000345672.5;etc;etc;etc do not have any matching peptide other than SGVYQHVTGEMMGGHAIR.
We could take a risk and assume that ENSP00000345672.5;etc;etc;etc are not present.

That was my idea.

Thanks for your comment and big thanks for OpenMS in general.

**Reply by jpfeuffer** [reply]
I see what you mean but yes, we do that already. The remaining "unresolved" ones are really "indistinguishable" in the sense that they share exactly the same set of observed peptides.
One *could* say that one is a little bit more likely based on its overall size or the location of the peptides but this is very hard to quantify and not too reliable.

**Reply by jpfeuffer** [reply]
Thanks for being so patient when trying OpenMS 😅
